test/library/packages/UnitTest/UnitTest.chpl

```diff
@@ -771,41 +821,62 @@ module UnitTest {
   /*Runs the tests*/
   proc runTest(tests: argType ...?n) throws {
 
+    var runAllTests = true;
+    var testNameList: [1..0] string;
+    if testNames != "None" {
+      runAllTests = false;
+      for test in testNames.split(" ") do testNameList.push_back(test.strip());
+    }
     // Assuming 1 global test suite for now
     // Per-module or per-class is possible too
     var testSuite = new TestSuite();
     testSuite.addTests(tests);
     var testResult = new TextTestResult();
-    // if skipId == 0 then
-    //   stdout.writeln("Found "+testSuite.testCount+" "+printTest(testSuite.testCount));
     for indx in (skipId+1)..testSuite.testCount {
       var test = testSuite[indx];
       try {
-        // Create a test object per test
-        var testObject = new Test();
-        //test is a FCF:
-        testResult.startTest(test: string, indx);
-        test(testObject);
-        testResult.addSuccess(test: string);
+        if runAllTests {
+          // Create a test object per test
+          var testObject = new Test();
```

Comment:
These four lines could maybe be a helper function

Reply:
Yes ma'am. Thanks for the feedback. I created a helper function.